### PR TITLE
Update resource/vocab markdown on script fields when cloning

### PR DESCRIPTION
**Pull request by bethanyaconnor**
Builds off of #43855 by updating the markdown in script fields, in particular the various description fields, on clone. When cloning lessons, we can store the cloned resource and vocab, but in this part of the code we don't have access to that. Instead, we rely on the logic in `copy_to_course_version`, which already looks if the resource/vocab already exists in the destination course version, to find the cloned object. These fields are also different than the ones updated on lesson clone as they are stored in `config/en.yml` as opposed to on the object.

I also found a pretty major bug in the logic updating this syntax in lesson clone, which I've fixed in 322d79ec31c72495b118bbbf79bf228305859331.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/app/models/lesson.rb [hunk 1]
```diff
@@ -872,12 +872,12 @@ def copy_to_unit(destination_unit, new_level_suffix = nil)
...
     update_resource_link_on_clone = proc do |resource|
       new_resource = copied_resource_map[resource.key] || resource.copy_to_course_version(course_version)
-      new_resource ? Services::GloballyUniqueIdentifiers.build_resource_key(new_resource) : Services::GloballyUniqueIdentifiers.build_resource_key(resource)
+      "[r #{new_resource ? Services::GloballyUniqueIdentifiers.build_resource_key(new_resource) : Services::GloballyUniqueIdentifiers.build_resource_key(resource)}]"
     end
 
     update_vocab_definition_on_clone = proc do |vocab|
       new_vocab = copied_vocab_map[vocab.key] || vocab.copy_to_course_version(course_version)
-      new_vocab ? Services::GloballyUniqueIdentifiers.build_vocab_key(new_vocab) : Services::GloballyUniqueIdentifiers.build_vocab_key(vocab)
+      "[v #{new_vocab ? Services::GloballyUniqueIdentifiers.build_vocab_key(new_vocab) : Services::GloballyUniqueIdentifiers.build_vocab_key(vocab)}]"
```
Comment (davidsbailey) [inline dashboard/app/models/lesson.rb:881]: nice catch!

File: dashboard/test/models/script_test.rb [hunk 2]
```diff
@@ -2859,10 +2859,10 @@ class MigratedScriptCopyTests < ActiveSupport::TestCase
       Script.stubs(:merge_and_write_i18n)
 
       @standalone_unit = create :script, is_migrated: true, is_course: true, version_year: '2021', family_name: 'csf', name: 'standalone-2021'
-      create :course_version, content_root: @standalone_unit
+      create :course_version, :with_course_offering, content_root: @standalone_unit
 
       @unit_group = create :unit_group
-      create :course_version, content_root: @unit_group
+      create :course_version, :with_course_offering, content_root: @unit_group
```
Comment (davidsbailey) [inline dashboard/test/models/script_test.rb]: will https://github.com/code-dot-org/code-dot-org/pull/43955 eliminate the need for this?

File: dashboard/test/models/script_test.rb [hunk 3]
```diff
@@ -2875,6 +2875,51 @@ class MigratedScriptCopyTests < ActiveSupport::TestCase
...
+          'data' => {
+            'script' => {
+              'name' => {
+                'new_name' => {
+                  'title' => '',
+                  'description_short' => "Description short: Resource: [r #{Services::GloballyUniqueIdentifiers.build_resource_key(copied_resource)}]. Vocab: [v #{Services::GloballyUniqueIdentifiers.build_vocab_key(copied_vocab)}].",
+                  'description_audience' => "Description audience: Resource: [r #{Services::GloballyUniqueIdentifiers.build_resource_key(copied_resource)}]. Vocab: [v #{Services::GloballyUniqueIdentifiers.build_vocab_key(copied_vocab)}].",
+                  'description' => "Description: Resource: [r #{Services::GloballyUniqueIdentifiers.build_resource_key(copied_resource)}]. Vocab: [v #{Services::GloballyUniqueIdentifiers.build_vocab_key(copied_vocab)}].",
+                  'student_description' => "Student description: Resource: [r #{Services::GloballyUniqueIdentifiers.build_resource_key(copied_resource)}]. Vocab: [v #{Services::GloballyUniqueIdentifiers.build_vocab_key(copied_vocab)}].",
+                  'lessons' => {}
```
Comment (davidsbailey) [inline dashboard/test/models/script_test.rb:2906]: I'm not sure if there is a real risk here, but from a readability standpoint it's a bit hard to convince myself that the vocab key generated by `build_vocab_key(copied_vocab)` is in fact the right expected value. would you consider  using the plaintext string for the resource/vocab keys here, or maybe asserting that the output of `build_vocab_key(copied_vocab)` is what you expect?

**Review by davidsbailey — APPROVED** [review]
LGTM after considering comments.

File: dashboard/test/models/script_test.rb (hunk 2 above)
Comment (bethanyaconnor) [reply dashboard/test/models/script_test.rb]: Yes -- I'll update. Thanks for catching!

File: dashboard/test/models/script_test.rb (hunk 3 above)
Comment (bethanyaconnor) [reply dashboard/test/models/script_test.rb:2906]: That makes sense. The key of the copied resource/vocab isn't necessarily deterministic, so it's a bit tough to check in plain text. I'll take a stab at making these more readable

Comment (davidsbailey) [reply dashboard/test/models/script_test.rb:2906]: I'm not sure what's best here, but perhaps just checking that the version year has changed would be enough? I think the pitfall which comes to mind is that somehow `build_vocab_key(copied_vocab)` is the same as `build_vocab_key(vocab)` and therefore the markdown also does not change.

Comment (davidsbailey) [reply dashboard/test/models/script_test.rb:2906]: the updates look great, thank you for tackling that!

**Pull request merged**